creator/referral_tokens/schema.py

```diff
@@ -0,0 +1,226 @@
+import graphene
+import django_filters
+from graphene import relay, ObjectType, Field
+from graphene_django import DjangoObjectType
+from graphene_django.filter import DjangoFilterConnectionField
+from django_filters import OrderingFilter
+from graphql_relay import from_global_id
+from botocore.exceptions import ClientError
+from django.conf import settings
+from graphql import GraphQLError
+from datetime import datetime, timedelta
+from .models import ReferralToken
+from creator.studies.models import Study
+from django.contrib.auth.models import Group
+from creator.events.models import Event
+from creator.users.schema import UserNode
+
+
+class ReferralTokenNode(DjangoObjectType):
+    is_valid = graphene.Boolean(source="is_valid")
+
+    class Meta:
+        model = ReferralToken
+        interfaces = (relay.Node,)
+        filter_fields = ()
+
+    @classmethod
+    def get_node(cls, info, uuid):
+        """
+        Only return if the user is allowed to view referral token
+        """
+        user = info.context.user
+
+        if not (user.has_perm("referral_tokens.view_referraltoken")):
+            raise GraphQLError("Not allowed")
+
+        if user.has_perm("referral_tokens.view_referraltoken"):
```

Comment:
This may need to get more complicated if we allow others to view token beside admins.
If we let investigators make tokens, we will want them to see tokens that they created or include their studies, but nothing more.
This is good for now because it's much more restrictive.